src/main/java/com/codicesoftware/plugins/hudson/actions/CheckoutAction.java

```diff
@@ -21,85 +22,93 @@
 
     private CheckoutAction() { }
 
-    public static Workspace checkout(
-            PlasticTool tool,
-            FilePath workspacePath,
-            String selector,
-            boolean useUpdate)
+    public static Workspace checkout(PlasticTool tool, FilePath workspacePath, String selector, CleanupMethod cleanup)
             throws IOException, InterruptedException, ParseException {
         List<Workspace> workspaces = WorkspaceManager.loadWorkspaces(tool);
 
-        cleanOldWorkspacesIfNeeded(tool, workspacePath, useUpdate, workspaces);
+        deleteOldWorkspacesIfNeeded(tool, workspacePath, cleanup, workspaces);
 
-        if (!useUpdate && workspacePath.exists()) {
-            workspacePath.deleteContents();
-        }
-
-        return checkoutWorkspace(tool, workspacePath, selector, workspaces);
+        return checkoutWorkspace(tool, workspacePath, selector, cleanup, workspaces);
     }
 
-    private static Workspace checkoutWorkspace(
-            PlasticTool tool,
-            FilePath workspacePath,
-            String selector,
-            List<Workspace> workspaces) throws IOException, InterruptedException, ParseException {
-
-        Workspace workspace = findWorkspaceByPath(workspaces, workspacePath);
-
-        if (workspace == null) {
-            LOGGER.fine("Creating new workspace");
+    private static Workspace checkoutWorkspace(PlasticTool tool, FilePath workspacePath, String selector, CleanupMethod cleanup, List<Workspace> workspaces)
```

Comment:
I noticed that you changed some method signatures that have one parameter per line to have all parameters in the same line.

I think it's a better idea to avoid changing the style in this big PRs because it can add noise. We could discuss creating a PR specifically to unify style and so on, but I believe this style-only changes should be avoided if possible.

Besides, this line (and some others) exceed the 120-character limit!

Reply:
The final code style based on official Sun and Google conventions was planned to be in next PR which I wanted to discuss together with you after this was merged. Will revert this. Checkstyle schema in next PR will enforce whichever form we choose.

Please not we are using **160** character limit. You are saying about 120 characters but if you would run checkstyle on Maven verify phase you would see there is no problems with the code.

We are in XXI century. Let's not get medieval about character limits. 80 character enforcement are in my opinion a joke. 120 is good but with everyone now having Full HD monitors or even bigger ones we should not get crazy. Too narrow code is not friendly. Also huge on liners are not. Something to discuss in next PR.

Reply:
Thanks for rolling those back.

My request was not about being medieval -as you said-, but to make it easier for anyone to review code. For instance, when I open the GitHub website, line width is about 110 characters. We should avoid making people scroll horizontally to read/review PRs.

I agree that it will be a good idea to decide a common style in a future PR and get it enforced during the maven build process.

Reply:
Yes, I apologize and totally agree with you. Later after writing my comment I started to think of situations where actually IDE is not used and line wrapping might be missing. Unfortunately there still such cases exist (mainly text diff tools) so keeping it 120 is the best choice. GitHub web is also a good example and is kind of unexpected as HTML text can be nicely wrapped in the browsers.

What I meant was that some tools are still in the past century, not us :) But you are right, we should support such cases. :+1: